programs/middleware/check_pass/src/lib.rs

```diff
@@ -169,6 +195,7 @@ impl<'info> ExecuteMiddleware<'info> {
             CheckPass::SEED_PREFIX,
             authority_key.as_ref(),
             gatekeeper_network_key.as_ref(),
+            ctx.accounts.middleware_account.failsafe.as_ref().map(|f| f.as_ref()).unwrap_or(&[0u8; 32]),
```

Comment:
Don't completely understand this handling of a Option<Pubkey>. Why can you not use unwrap_or directly?

Reply:
In summary, I need both as_refs.

- as_rev 1 turns the &Option<Pubkey> to an Option<&Pubkey>
- This allows the map to map the option and return (using as_ref again) the internal pubkey as an u8 slice (if I don't do that, map still works but you get moving issues)

an alternative would be to do unwrap_or(Pubkey(0u8; 30)) (or similar) but that would not pass as 0000... is not a valid pubkey. So you have to extract it into a byte slice first before doing the unwrap.